src/QMCDrivers/DMC/WalkerControl.h

```diff
@@ -68,7 +68,22 @@ class WalkerControl : public MPIObjectBase
    */
   inline void setTrialEnergy(FullPrecRealType et) { trial_energy_ = et; }
 
-  /** unified: perform branch and swap walkers as required 
+  /** unified: perform branch and swap (balance) walkers as required
+   *  **This has many side effects**
+   *  ## For:
+   *  ### dynamic population
+   *  1. compute multiplicity. If iter 0 and all of warmup -> multiplicity = 1
+   *     Multiplicity in normal branching is walker->Weight + rng()
+   *  2. compute curData, collect multiplicity on every rank
+   *  ### fixed population
+   *  1. compute curData, collect weight on every rank
+   *  2. compute multiplicity by comb method
+   *  ---
+   *  3. figure out final distribution, apply walker count ceiling
+   *  4. collect good, bad walkers
+   *  5. communicate walkers
+   *  6. unpack received walkers, apply walker count floor
+   *  7. call MCPopulation to amplify walkers with Multiplicity > 1
```

Comment:
Please move this back to its original location. Users calling this API doesn't need to know the implementation detail.

Reply:
returned.  I left some information for this call in the header.